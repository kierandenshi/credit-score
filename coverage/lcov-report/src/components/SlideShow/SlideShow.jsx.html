<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/SlideShow/SlideShow.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/SlideShow</a> SlideShow.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.44% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>15/25</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Transition from 'react-transition-group/Transition';
import TransitionGroup from 'react-transition-group/TransitionGroup';
import {
    schedule,
    deschedule,
} from '@middleware/scheduledActionMiddleware';
import { showNextSlide } from '@redux/slideshow';
import './_SlideShow.scss';
&nbsp;
const SlideTransition = <span class="fstat-no" title="function not covered" >({</span> children, duration, in: inProp }) =&gt; {
    const defaultStyle = <span class="cstat-no" title="statement not covered" >{</span>
        transition: `${duration}ms`,
        transitionProperty: 'left',
        position: 'absolute',
    };
&nbsp;
    const transitionStyles = <span class="cstat-no" title="statement not covered" >{</span>
        entering: { left: '414px' },
        entered: { left: '0px' },
        exiting: { left: '-414px' },
        exited: { left: '0px' },
    };
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
        &lt;Transition
            in={inProp}
            timeout={{
                enter: duration,
                exit: duration,
            }}
        &gt;
            {
<span class="fstat-no" title="function not covered" >                (s</span>tatus) =&gt; {
<span class="cstat-no" title="statement not covered" >                    if (status === 'exited') {</span>
<span class="cstat-no" title="statement not covered" >                        return null;</span>
                    }
<span class="cstat-no" title="statement not covered" >                    return React.cloneElement(children, {</span>
                        style: { ...defaultStyle, ...transitionStyles[status] },
                    });
                }
            }
        &lt;/Transition&gt;
    );
};
&nbsp;
export class SlideShow extends Component {
    constructor(props) {
        super(props);
        const { children } = props;
        this.slides = children ? children.length : <span class="branch-1 cbranch-no" title="branch not covered" >0;</span>
        this.delay = 10;
    }
&nbsp;
    componentDidMount() {
        const { schedule } = this.props; // eslint-disable-line no-shadow
        this.slides &gt; 1 &amp;&amp; schedule('slideshowTick', { action: showNextSlide(this.slides), every: this.delay });
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    co</span>mponentWillUnmount() {
        const { deschedule } = <span class="cstat-no" title="statement not covered" >this.props;</span> // eslint-disable-line no-shadow
<span class="cstat-no" title="statement not covered" >        deschedule('slideshowTick');</span>
    }
&nbsp;
    skipSlide = () =&gt; {
        const { schedule, deschedule } = this.props; // eslint-disable-line no-shadow
        deschedule('slideshowTick');
        this.props.showNextSlide(this.slides);
        this.slides &gt; 1 &amp;&amp; schedule('slideshowTick', { action: showNextSlide(this.slides), every: this.delay });
    }
&nbsp;
    render() {
        const { children, currentSlide } = this.props;
&nbsp;
        return (
            /* eslint-disable */
            &lt;div className={'slide-show'} onClick={this.skipSlide} ref={<span class="fstat-no" title="function not covered" >(e</span>) =&gt; { <span class="cstat-no" title="statement not covered" >this.container = e; </span>}}&gt;
                &lt;TransitionGroup&gt;
                    &lt;SlideTransition in key={`slide${currentSlide}`} duration={500}&gt;
                        &lt;div&gt;{children[currentSlide]}&lt;/div&gt;
                    &lt;/SlideTransition&gt;
                &lt;/TransitionGroup&gt;
            &lt;/div&gt;
            /* eslint-enable */
        );
    }
}
&nbsp;
SlideShow.defaultProps = {
    children: null,
};
&nbsp;
SlideShow.propTypes = {
    children: PropTypes.node,
    currentSlide: PropTypes.number.isRequired,
    schedule: PropTypes.func.isRequired,
    deschedule: PropTypes.func.isRequired,
    showNextSlide: PropTypes.func.isRequired,
};
&nbsp;
export default connect(
<span class="fstat-no" title="function not covered" >    st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
        currentSlide: state.slideshow.currentSlide,
    }),
    {
        schedule,
        deschedule,
        showNextSlide,
    },
)(SlideShow);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 16 2018 09:27:06 GMT+0000 (GMT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
