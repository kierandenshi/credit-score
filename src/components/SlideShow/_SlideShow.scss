@import '~@css/includes/helpers';

.slide-show {
  position: relative;
  width: 298px;
  height: 298px;
  cursor: pointer;
  clip-path: circle(298px);

  @include media('>phone') {
    width: 414px;
    height: 414;
    cursor: pointer;
    clip-path: circle(414px);
  }

  &__slide {
    position: absolute;
    width: 100%;
    height: 100%;

    &.animate {
      transition: left 0.3s
    }

    


  }
}
