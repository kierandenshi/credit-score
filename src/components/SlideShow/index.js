export { default } from './SlideShow.jsx';
