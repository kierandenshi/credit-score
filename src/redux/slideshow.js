const createAction = type => payload => ({ type, payload });
const NEXT_SLIDE = 'NEXT_SLIDE';

export const showNextSlide = createAction(NEXT_SLIDE);

export const initialState = {
    currentSlide: 0,
};

export default (state = initialState, { type, payload }) => {
    switch (type) {
        case NEXT_SLIDE:
            return {
                ...state,
                currentSlide: state.currentSlide === payload - 1 ? 0 : state.currentSlide + 1,
            };

        default:
            return state;
    }
};
