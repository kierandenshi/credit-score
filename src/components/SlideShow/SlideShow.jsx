import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Transition from 'react-transition-group/Transition';
import TransitionGroup from 'react-transition-group/TransitionGroup';
import {
    schedule,
    deschedule,
} from '@middleware/scheduledActionMiddleware';
import { showNextSlide } from '@redux/slideshow';
import './_SlideShow.scss';

const SlideTransition = ({ children, duration, in: inProp }) => {
    const defaultStyle = {
        transition: `${duration}ms`,
        transitionProperty: 'left',
        position: 'absolute',
    };

    const transitionStyles = {
        entering: { left: '414px' },
        entered: { left: '0px' },
        exiting: { left: '-414px' },
        exited: { left: '0px' },
    };

    return (
        <Transition
            in={inProp}
            timeout={{
                enter: duration,
                exit: duration,
            }}
        >
            {
                (status) => {
                    if (status === 'exited') {
                        return null;
                    }
                    return React.cloneElement(children, {
                        style: { ...defaultStyle, ...transitionStyles[status] },
                    });
                }
            }
        </Transition>
    );
};

export class SlideShow extends Component {
    constructor(props) {
        super(props);
        const { children } = props;
        this.slides = children ? children.length : 0;
        this.delay = 10;
    }

    componentDidMount() {
        const { schedule } = this.props; // eslint-disable-line no-shadow
        this.slides > 1 && schedule('slideshowTick', { action: showNextSlide(this.slides), every: this.delay });
    }

    componentWillUnmount() {
        const { deschedule } = this.props; // eslint-disable-line no-shadow
        deschedule('slideshowTick');
    }

    skipSlide = () => {
        const { schedule, deschedule } = this.props; // eslint-disable-line no-shadow
        deschedule('slideshowTick');
        this.props.showNextSlide(this.slides);
        this.slides > 1 && schedule('slideshowTick', { action: showNextSlide(this.slides), every: this.delay });
    }

    render() {
        const { children, currentSlide } = this.props;

        return (
            /* eslint-disable */
            <div className={'slide-show'} onClick={this.skipSlide} ref={(e) => { this.container = e; }}>
                <TransitionGroup>
                    <SlideTransition in key={`slide${currentSlide}`} duration={500}>
                        <div>{children[currentSlide]}</div>
                    </SlideTransition>
                </TransitionGroup>
            </div>
            /* eslint-enable */
        );
    }
}

SlideShow.defaultProps = {
    children: null,
};

SlideShow.propTypes = {
    children: PropTypes.node,
    currentSlide: PropTypes.number.isRequired,
    schedule: PropTypes.func.isRequired,
    deschedule: PropTypes.func.isRequired,
    showNextSlide: PropTypes.func.isRequired,
};

export default connect(
    state => ({
        currentSlide: state.slideshow.currentSlide,
    }),
    {
        schedule,
        deschedule,
        showNextSlide,
    },
)(SlideShow);
