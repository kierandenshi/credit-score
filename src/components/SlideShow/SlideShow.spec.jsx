import React from 'react';
import { shallow } from 'enzyme';
import { getProps, updateProps } from '@test/helpers';
import { showNextSlide } from '@redux/slideshow';
import { SlideShow } from './SlideShow.jsx';

const props = {
    currentSlide: 0,
    children: [<div id="child1" />, <div id="child2" />],
    schedule: jest.fn(),
    deschedule: jest.fn(),
    showNextSlide: jest.fn(),
};

describe('SlideShow component', () => {
    const wrapper = shallow(<SlideShow {...getProps(props)} />);
    const { slides, delay } = wrapper.instance();

    it('should render', () => {
        expect(wrapper.find('.slide-show')).toMatchSnapshot();
    });

    it('should start the animation scheduler', () => {
        expect(props.schedule).toHaveBeenCalledWith(
            'slideshowTick',
            { action: showNextSlide(slides), every: delay },
        );
    });

    it('should render the correct child element', () => {
        expect(wrapper.find('#child1')).toHaveLength(1);
        updateProps(wrapper, props, { currentSlide: 1 });
        expect(wrapper.find('#child2')).toHaveLength(1);
    });

    it('should skip to next slide on click', () => {
        wrapper.simulate('click');
        expect(props.deschedule).toHaveBeenCalledWith('slideshowTick');
        expect(props.schedule).toHaveBeenCalledWith(
            'slideshowTick',
            { action: showNextSlide(slides), every: delay },
        );
        expect(props.showNextSlide).toHaveBeenCalled();
    });
});
