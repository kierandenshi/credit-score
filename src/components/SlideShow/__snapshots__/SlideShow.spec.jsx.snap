// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideShow component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <SlideShow
      currentSlide={0}
      deschedule={[MockFunction]}
      schedule={
        [MockFunction] {
          "calls": Array [
            Array [
              "slideshowTick",
              Object {
                "action": Object {
                  "payload": 2,
                  "type": "NEXT_SLIDE",
                },
                "every": 10,
              },
            ],
          ],
        }
      }
      showNextSlide={[MockFunction]}
    >
      <div
        id="child1"
      />
      <div
        id="child2"
      />
    </SlideShow>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TransitionGroup
          childFactory={[Function]}
          component="div"
        >
          <SlideTransition
            duration={500}
            in={true}
          >
            <div>
              <div
                id="child1"
              />
            </div>
          </SlideTransition>
        </TransitionGroup>,
        "className": "slide-show",
        "onClick": [Function],
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "childFactory": [Function],
          "children": <SlideTransition
            duration={500}
            in={true}
          >
            <div>
              <div
                id="child1"
              />
            </div>
          </SlideTransition>,
          "component": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "slide0",
          "nodeType": "function",
          "props": Object {
            "children": <div>
              <div
                id="child1"
              />
            </div>,
            "duration": 500,
            "in": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                id="child1"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "child1",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TransitionGroup
            childFactory={[Function]}
            component="div"
          >
            <SlideTransition
              duration={500}
              in={true}
            >
              <div>
                <div
                  id="child1"
                />
              </div>
            </SlideTransition>
          </TransitionGroup>,
          "className": "slide-show",
          "onClick": [Function],
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "childFactory": [Function],
            "children": <SlideTransition
              duration={500}
              in={true}
            >
              <div>
                <div
                  id="child1"
                />
              </div>
            </SlideTransition>,
            "component": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "slide0",
            "nodeType": "function",
            "props": Object {
              "children": <div>
                <div
                  id="child1"
                />
              </div>,
              "duration": 500,
              "in": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  id="child1"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "child1",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <SlideTransition
          duration={500}
          in={true}
        >
          <div>
            <div
              id="child1"
            />
          </div>
        </SlideTransition>
      </TransitionGroup>,
      "className": "slide-show",
      "onClick": [Function],
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "childFactory": [Function],
        "children": <SlideTransition
          duration={500}
          in={true}
        >
          <div>
            <div
              id="child1"
            />
          </div>
        </SlideTransition>,
        "component": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "slide0",
        "nodeType": "function",
        "props": Object {
          "children": <div>
            <div
              id="child1"
            />
          </div>,
          "duration": 500,
          "in": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              id="child1"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "child1",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TransitionGroup
          childFactory={[Function]}
          component="div"
        >
          <SlideTransition
            duration={500}
            in={true}
          >
            <div>
              <div
                id="child1"
              />
            </div>
          </SlideTransition>
        </TransitionGroup>,
        "className": "slide-show",
        "onClick": [Function],
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "childFactory": [Function],
          "children": <SlideTransition
            duration={500}
            in={true}
          >
            <div>
              <div
                id="child1"
              />
            </div>
          </SlideTransition>,
          "component": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "slide0",
          "nodeType": "function",
          "props": Object {
            "children": <div>
              <div
                id="child1"
              />
            </div>,
            "duration": 500,
            "in": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                id="child1"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "child1",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
